Source/Core/Common/Arm64Emitter.cpp

```diff
@@ -3039,28 +3039,31 @@ void ARM64FloatEmitter::FMOV(ARM64Reg Rd, ARM64Reg Rn, bool top)
   {
     EmitScalar1Source(0, 0, IsDouble(Rd), 0, Rd, Rn);
   }
+  else if (IsGPR(Rd) != IsGPR(Rn))
+  {
+    const ARM64Reg gpr = IsGPR(Rn) ? Rn : Rd;
+    const ARM64Reg fpr = IsGPR(Rn) ? Rd : Rn;
+
+    const int sf = Is64Bit(gpr) ? 1 : 0;
+    const int type = Is64Bit(gpr) ? (top ? 2 : 1) : 0;
+    const int rmode = top ? 1 : 0;
+    const int opcode = IsGPR(Rn) ? 7 : 6;
+
+    ASSERT_MSG(DYNA_REC, !top || IsQuad(fpr), "FMOV: top can only be used with quads");
+
+    // Should this check be more lenient? Sometimes you do want to do things like
```

Comment:
This really seems like a TODO

Reply:
Yes, it is that type of comment.

Reply:
Well, then placing "TODO:" before the comment would make it easier to recognize as such.

Reply:
I thought it was fine like this, but alright, I've added "TODO:" in front of it.